vhdl_lang/src/named_entity/design.rs

```diff
@@ -22,7 +22,13 @@ pub enum Design<'a> {
     Package(Visibility<'a>, Region<'a>),
     PackageBody,
     UninstPackage(Visibility<'a>, Region<'a>),
-    PackageInstance(Region<'a>),
+    // The second parameter of `PackageInstance` is `true`, when this package instance is
+    // part of a generic map, i.e.,
+    // generic map (
+    //     package foo is new bar generic map (<>)
+    // );
+    // otherwise, the instantiated packages appears regularly in the declarative region.
+    PackageInstance(Region<'a>, bool),
```

Comment:
I think it would be clearer to have a separate InterfacePackage or simial as a new enum variant instead of having an extra boolean.